lib/galaxy/webapps/galaxy/controllers/async.py

```diff
@@ -52,30 +50,62 @@ def index(self, trans, tool_id=None, data_secret=None, **kwd):
         if not tool:
             return f"Tool with id {tool_id} not found"
 
-        #
-        # we have an incoming data_id
-        #
         if data_id:
-            if not URL:
-                return f"No URL parameter was submitted for data {data_id}"
+            #
+            # we have an incoming data_id
+            #
             data = trans.sa_session.query(trans.model.HistoryDatasetAssociation).get(data_id)
```

Comment:
Another thing we noticed is that this `data_id` is circulating unencoded between requests. It should be encoded right?

Reply:
Ideally, but I guess this is somewhat internal ?

Reply:
Yeah, it looks pretty internal, maybe when this gets translated to a proper API endpoint (if it should at some point) we can revisit the encoding.